### PR TITLE
Deb Package

### LINKED ISSUE #143: feat:  Publish .deb packages
Hi,

could you perhaps publish .deb packages for Debian based distros, so eza could be easily installed on Debian or potentially included in Debian repo or added to deb-get:
https://github.com/wimpysworld/deb-get

### PR DESCRIPTION
**Pull request by gierens**
This adds the script `devtools/deb-package.sh` to build a `.deb` package from the build release binary and man pages. It adds the deb package to `.gitingore` and also replace `exa` with `eza` there.

It also adds a name section to the `eza_colors-explanation` man page, as this is according to the Debian linter `lintian` good practice to have it as first section.

It finally adds a section to the `README.md` with instructions to install it from a repo I created and adds the corresponding GPG public key with which the package is signed to `deb.asc`.

Note, in case you test this the instructions assume that `deb.asc` is already in the `main` branch of this repo, so adjust the url for the initial `wget` to `https://raw.githubusercontent.com/gierens/eza/deb-package/deb.asc`.

Screenshot from 2023-08-18 13-38-55

Resolves #143

### CODE REVIEW

**Comment by cafkafk** [comment]
I'm not on debian, so maybe @sbstn87 would be willing to give this a spin and report back how it went?

**Comment by PThorpe92** [comment]
Everything looks good, I cloned your fork/branch onto my work laptop runs Ubuntu.

I did test this out to install with the exact instructions in the README, 
I replaced the url in the initial `wget` with the link to your fork, and copy + pasted the commands from the readme. 

```
E: The repository 'http://deb.gierens.de lunar Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
```
Is there something I am missing? I know the PR mentioned deb-get, which I have installed but I would imagine that is waiting on the release to be hosted in the main repo.

**Review by cafkafk — CHANGES_REQUESTED** [review]
This PR looks very high quality and polished, well done!

Consider this an approval, I'll just give something like a day for potential testers to try this out, if no one tries it, I'll approve tomorrow morning.

**Comment by gierens** [comment]
> ```
> E: The repository 'http://deb.gierens.de lunar Release' does not have a Release file.
> N: Updating from such a repository can't be done securely, and is therefore disabled by default.
> ```
Ah, yeah, my bad, no you're not missing anything ... I tried to make the repo superfancy with different subfolders for different distros, and `lunar` is just not among the list I put in ... Since we just have a static binary and don't need to take care of different dependencies for different distros, we don't need it anyway ... I'll simplify it later or tomorrow. Thank's for the hint.

File: README.md [hunk 1]
```diff
@@ -103,6 +103,17 @@ For `nix-env` users:
 nix-env -i eza
 ```
 
+### Debian and Ubuntu
+Eza is available from [deb.gierens.de](http://deb.gierens.de). The GPG public
```
[the remark this answers is not in the document]
Comment (gierens) [reply README.md:107]: Great, wouldn't be a big deal if not, I wanted to know how to set up an apt repo anyway, so wasn't any wasted effort on my side ;) ... Once eza has an own domain and stuff, we could also change it, no problem :)

File: devtools/deb-package.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+Upstream-Contact: Christina Sørensen <christina@cafkafk.com>
+Source: https://github.com/eza-community/eza/releases
+
+Files: *
+License: MIT
+Copyright: 2023 Christina Sørensen <christina@cafkafk.com>
+
+Files: debian/*
+License: GPL-3+
+Copyright: 2023 Christina Sørensen <christina@cafkafk.com>
```
[the remark this answers is not in the document]
Comment (gierens) [reply devtools/deb-package.sh:49]: So this is just for the `debian/` subfolder of the repo ... but interesting, I thought I had changed that to MIT as well ... Hm, I don't have a strong opinion on that.

**Comment by gierens** [comment]
@cafkafk I'll write, once I'm done with the changes to the repo and so on ;)

**Comment by gierens** [comment]
@cafkafk Alright, so I restructured the repo to just a `stable` subfolder instead of different ones for different distros and updated the `README.md` accordingly. So I adjusted the commands to ignore the distro and just go to `stable`, and also to put the repo information into a separate file under `/etc/apt/sources.list.d/` instead of appending to `/etc/apt/sources.list`. This makes it more manageable.

I fixed the left-over GPL license, it's now MIT for the entire `deb` package. For easy merging I already rebase my branch on the current state of `main`.

To test this use:
```bash
wget -qO- https://raw.githubusercontent.com/gierens/eza/deb-package/deb.asc | sudo tee /etc/apt/trusted.gpg.d/gierens.asc
echo "deb http://deb.gierens.de stable main" | sudo tee /etc/apt/sources.list.d/gierens.list
sudo apt update
sudo apt install -y eza
```
which is basically the instructions from the readme just with my repo and branch.

In case one already installed it before these changes prepend the following commands:
```bash
sudo apt purge eza
sudo sed -i '/gierens/d' /etc/apt/sources.list
sudo apt update
```

**Comment by sbstn87** [comment]
I have installed the .deb package by following the instructions above and it seems to work fine for me. 

The only thin I have noticed, is that it took a lot of time to update the repo the first time. So I have uninstalled eza and installed it again - this time it was quick and smooth.

**Comment by gierens** [comment]
> I have installed the .deb package by following the instructions above and it seems to work fine for me.

Great, nice to hear.

> The only thin I have noticed, is that it took a lot of time to update the repo the first time. So I have uninstalled eza and installed it again - this time it was quick and smooth.

Ok, that's good to know, I'll keep an eye on it and move repo to a different machine if it necessary.

**Comment by PThorpe92** [comment]
Ran those commands and it installed just fine on my Ubuntu machine :+1: good stuff!

**Comment by cafkafk** [comment]
> As a side note, we break 2000 commits then 🥳

Let's goooooooooooo!!! :tada: 

I'm super tired right now, so I'll look at merging it tomorrow so I don't make any mistakes, seems like it's working for everyone thou so it should be smooth sailing!

**Review by cafkafk — APPROVED** [review]
This seems to work for users, and not have any overlap with any critical code, well done, LGTM :+1:

**Pull request merged**